### PR TITLE
feat: contract readonly method rpc

### LINKED ISSUE #3689: [Demand] add rpc param to useContract()
## Catalogue

- [ ] UI/UX
- [ ] Mask Core Feature
- [ ] Mask Plugin Feature
- [ ] Security/Cryptography Feature
- [ ] Other
  - /_List it here_/

## Part 1: Introduction and Background
The RPC for contract is equal to the current one of wallet provider.

## Part 2: Problem to Solve
Now we want to read contract data from any chain no matter which chain you are on.

## Part 3: Expected Behavior
E.g. Metamask is on eth-mainnet, BSC ITO card is still visible.

### LINKED ISSUE #3765: [Bug] Change the unit for ito infor table's balance
### How to reproduce?

1.Post a ITO;
2.Find the ito information table in Past tab of ITO;
3.Check the balance column of information table.

### What happened?
The unit of balance is the token swapped to.
截屏2021-07-27 下午1 27 04

### What should happen?
The unit of balance should be the token swapped from.

### Mask Version:

- [x] (Browser) Installed from the extension store (version: )
- [ ] (App) Installed from the app store (version: )
- [ ] master branch
- [ ] Build of version/commit hash:

(Please write down build command/flags if there is anything special)

### System

 
- [ ] Windows 10
- [x] Mac OS X Catalina
- [ ] Linux: 

### Platform/Browser

 
- [x] Latest Chrome (stable channel)
- [ ] Latest Firefox (stable channel)
- [ ] Android 11
- [ ] iOS 14

### PR DESCRIPTION
**Pull request by zhouhanseng**
There's only one readonly method in rp and ito: check_availability()

closes #3689
closes #3765

### CODE REVIEW

**Comment by zhouhanseng** [comment]
image

File: packages/maskbook/src/web3/UI/EthereumChainBoundary.tsx [hunk 1]
```diff
@@ -29,10 +32,11 @@ export function EthereumChainBoundary(props: EthereumChainBoundaryProps) {
     const chainId = useChainId()
     const providerType = useValueRef(currentProviderSettings)
 
+    const { noSwitchNetworkTip = false } = props
     const expectedChainId = props.chainId
-    const expectedNetwork = getChainName(expectedChainId)
+    const expectedNetwork = expectedChainId === ChainId.BSC ? 'BSC' : getChainName(expectedChainId)
```
Comment (zhouhanseng) [inline packages/maskbook/src/web3/UI/EthereumChainBoundary.tsx:37]: ming thinks 'Swtich to BSC Network' is better than ' Swtich to Binance Smart Chain Mainnet Network'

File: packages/maskbook/src/plugins/ITO/SNSAdaptor/hooks/useClaimablePools.ts [hunk 2]
```diff
@@ -24,65 +11,28 @@ export interface SwappedToken {
...
-    const chainId = useChainId()
-    const { ITO_CONTRACT_ADDRESS } = useITOConstants()
-    // Todo: Remove the code after the period that old ITO is being used and continues to be used for a while
-    const { contract: ITO_Contract } = useITO_Contract(isMainnetOld ? ITO_CONTRACT_ADDRESS : undefined)
-
+    const { ITO_CONTRACT_ADDRESS, ITO2_CONTRACT_ADDRESS } = useITOConstants(chainId)
+    const contractAddress = isMainnetOld ? ITO_CONTRACT_ADDRESS : ITO2_CONTRACT_ADDRESS
     //#region fetch claimable pool
-    const { value: poolsFromSubgraph = [], loading: loadingSubgraph } = useClaimablePoolsBySubgraph()
-    const { value: poolsFromWeb3 = [], loading: loadingWeb3 } = useClaimablePoolsByWeb3()
```
Comment (zhouhanseng) [inline packages/maskbook/src/plugins/ITO/SNSAdaptor/hooks/useClaimablePools.ts:36]: getPastLogs() through our own rpc is much slower ( > 1 min on mainnet) and unstable (lots of error e.g. CORS on bsc and matic) than thegraph. Remove it temporaily.

Better solution in the future: send a preflight request to thegraph, if it is down, then fallback to `web3.getPastLogs()`

File: packages/shared/src/wallet/components.tsx [hunk 3]
```diff
@@ -1,17 +1,33 @@
...
     significant?: number
     symbol?: string
 }
 
-export const FormattedBalance: FC<FormattedBalanceProps> = ({ value, decimals, significant, symbol }) => {
+export const FormattedBalance: FC<FormattedBalanceProps> = (props) => {
+    const { value, decimals, significant, symbol, classes } = props
     const formatted = formatBalance(value, decimals, significant)
-    if (symbol) return <Fragment>{`${formatted} ${symbol}`}</Fragment>
+    const ordinaryClasses = useStyles()
```
Comment (zhouhanseng) [inline packages/shared/src/wallet/components.tsx:23]: I will move `useStylesExtends()` to `@masknet/shared` in another PR.

**Comment by wenluomask** [comment]
Checked.

**Review by nuanyang233 — APPROVED** [review]

**Pull request merged**